homeassistant/components/nest/legacy/__init__.py

```diff
@@ -196,7 +197,7 @@ def set_eta(service):
                     )
                     structure.away = AWAY_MODE_AWAY
 
-                    now = datetime.utcnow()
+                    now = dt_util.utcnow().replace(tzinfo=None)
```

Comment:
The python3.12 deprecation notice seems to say that the goal of the deprecation is to ensure that datetimes have a timezone specified to avoid bugs where a datetime without a timezone is interpreted as a local time.

Given that goal, while removing the timezone is safer given that it is a no-op, it still means there is cleanup work here to do.  Should we be more aggressive about cleaning these up now or track them somehow to fix later?

Reply:
It seems like if we're doing this, it's not really meeting the spirit of the deprecation.  An alternative is perhaps the cleanup should happen in a second step, leaving the deprecated method in place until the underlying root issue is resolved (e.g. if it needs a code owner to take a closer look)

Reply:
I modified a subset of the cases here to use an aware object where it was easy to determine that doing so would work, but left naive ones in place where it wasn't immediately clear to me if not.

There are some cases where the naive one is explicitly required per comments, and some others in which the object gets formatted to a string -- depends on the consumer if presence of timezone there (which would happen for aware objects) would cause breakage or not.

But yeah, I agree there's some cleanup work left to do. I'm open to suggestions how to handle that. TODO comments would be one possibility (along with a linter exclusion), another one would be to just leave such cases unmodified and handle them $sometime later. Or we could try to beat them into shape already in this PR.

Reply:
Went through and cleared a few aware/naive issues, this one and the ones marked with comments below are the remaining cleanups that I can tell. I suppose it wouldn't be too hard to figure them out while in this PR still.